fuzzers/forkserver/forkserver_simple/src/main.rs

```diff
@@ -166,17 +169,30 @@ pub fn main() {
 
     let observer_ref = edges_observer.handle();
 
+    // By deisgn use observers to collect stdout/stderr to illustrate usages
+    let stdout = StdOutObserver::new("simple_stdout");
+    let stderr = StdErrObserver::new("simple_stderr");
+
+    let stdout_handle = stdout.handle();
+    let stderr_handle = stderr.handle();
+
     let mut tokens = Tokens::new();
     let mut executor = ForkserverExecutor::builder()
         .program(opt.executable)
-        .debug_child(debug_child)
         .shmem_provider(&mut shmem_provider)
         .autotokens(&mut tokens)
         .parse_afl_cmdline(args)
         .coverage_map_size(MAP_SIZE)
         .timeout(Duration::from_millis(opt.timeout))
-        .kill_signal(opt.signal)
-        .build(tuple_list!(time_observer, edges_observer))
+        .kill_signal(opt.signal);
+
+    if debug_child {
```

Comment:
this is ugly. can you not hide this inside forkserver executor's code?

Reply:
This is intentionally implementing `debug_child` by the new functionalities proposed by this PR.